R/uvhydrograph-render.R

```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
   
   return(y.endpoint)
 }
+
+
+#' TODO
+#' @importFrom grDevices rgb
+getPrimaryPlotConfig <- function(reportObject, info, correctionLabels, dataSides, dataLimits) {
+  styles <- getUvStyles()
+  
+  x <- reportObject[[1]]$time
+  y <- reportObject[[1]]$value
+  
+  comp_lbl <- info$comp_UV_lbl
+  comp_type <- info$comp_UV_type
+  legend.name <- reportObject[[1]]$legend.name
+  plotStartDate <- info$plotStartDate
+  plotEndDate <- info$plotEndDate
+  
+  corrArrowPositions <- getCorrectionArrowPositions(x, y, correctionLabels)
+  corrAblinePositions <- getCorrectionAbLines(reportObject)
+  
+  primary_lbl <- info$primary_lbl
+  reference_lbl <- info$reference_lbl
+  comp_lbl <- paste("Comparison", info$comp_UV_TS_lbl, "@", info$comp_UV_lbl)
+  
+  compAxes <- TRUE
+  compAnnotations <- TRUE
+  compLabel <- primary_lbl
+  
+  if(dataSides$comparison == 6){
```

Comment:
I wrote this part back when adding the third Y-Axis. This is the logic for specifying how to label each side. If any TS are plotted on the same side they should have the same label, but if they'r eon different sides they should use different labels. So this is checking what side each TS is being plotted on and applying the proper label. I can add some comments to clarify the logic in a separate PR after this is merged.